lib/trusted-mpt-light-client-types/src/signed_data.rs

```diff
@@ -0,0 +1,45 @@
+use unionlabs::{
+    encoding::{Encode, Encoding},
+    primitives::{H256, H512},
+};
+
+#[derive(Debug, Clone, PartialEq)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "bincode", derive(bincode::Encode, bincode::Decode))]
+pub struct SignedData<T> {
+    data: T,
+    signature: H512,
+}
+
+pub struct SignatureVerificationFailure;
```

Comment:
impl Error for this (can derive thiserror::Error and use the #[error()] attribute)